src/Application.js

```diff
@@ -116,10 +117,17 @@ export default class Application extends Dispatcher {
   _setupConfig() {
     this.setUserConfig(null, _userConfig)
 
-    this.config = Object.assign(this.config, _defaultConfig)
-    this.config = Object.assign(this.config, new ConfigurationManager(this.config).getConfig())
-    this.config = Object.assign(this.config, this._opts)
-    if(typeof this.config.debug === 'undefined') this.config.debug = (!process.env.NODE_ENV || process.env.NODE_ENV == 'development')
+    return this.emit('config').then((config={}) => {
+      if (_.isArray(config)) {
+        config = Object.assign(...config)
```

Comment:
I'm not clear on how `config` is getting collected, but (1) do you need a `deepExtend()` here, and (2) is is safe to overwrite `config[0]` with the `Object.assign()`?

Reply:
yah, this should probably deepExtend for both points.